server/controllers/board.js

```diff
@@ -1,97 +1,149 @@
 const asyncHandler = require("express-async-handler");
 const Board = require("../models/Board");
-const { v4: uuidv4, v4 } = require("uuid");
+const User = require("../models/User");
 
-exports.getBoards = asyncHandler(async (req, res) => {
-  const boards = await Board.find();
+exports.getBoard = asyncHandler(async (req, res) => {
+  const board = await Board.findOne({ _id: req.params.id }).populate({
+    path: "columns",
+    populate: {
+      path: "cards",
+      model: "card",
+    },
+  });
 
-  return res.json(boards);
+  if (!board) {
+    res.status(404);
+    throw new Error("Board not found");
+  }
+
+  return res.status(200).json(board);
 });
 
-exports.getBoard = asyncHandler(async (req, res) => {
-  try {
-    const board = await Board.findById(req.params.id);
+exports.createBoard = asyncHandler(async (req, res) => {
+  const { name } = req.body;
+
+  const newBoard = await Board.create({
+    name,
+    user: req.params.id,
+  });
 
-    if (!board) res.status(404).json({ error: "Board not found" });
+  console.log(newBoard.name);
+  // For testing with postman
+  // const newBoard = await Board.create({
+  //   name,
+  //   user: "60b91df75edef24420936968",
+  // });
 
-    return res.status(200).json(board);
-  } catch (error) {
-    return res.json(error);
+  if (!newBoard) {
+    res.status(400);
+    throw new Error("Board could not be created.");
   }
+
+  await newBoard.createTemplateBoard();
+
+  const board = await Board.populate(newBoard, {
+    path: "columns",
+    populate: {
+      path: "cards",
+    },
+  });
+  return res.status(200).json({ board });
+
+  // { board?: a board, error?: string}
 });
 
-exports.createBoard = asyncHandler(async (req, res) => {
+exports.updateBoardName = asyncHandler(async (req, res) => {
+  const { id } = req.params;
   const { name } = req.body;
-  const newBoard = await Board.create({
-    name: name,
-    columns: [
-      {
-        name: "Not Started",
-        cards: [
-          {
-            _id: "car-1",
-            columnId: "col-1",
-            name: "Essay on the environment",
-            tag: "green",
-          },
-        ],
-        _id: v4(),
-        createdAt: Date.now(),
-      },
-      {
-        name: "In Progress",
-        cards: [
-          {
-            _id: "car-2",
-            columnId: "col-2",
-            name: "Midterm exam",
-            dueDate: new Date(),
-            tag: "red",
-          },
-          {
-            _id: "car-3",
-            columnId: "car-3",
-            name: "Homework",
-            tag: "red",
-          },
-        ],
-        _id: v4(),
-        createdAt: Date.now(),
+
+  const board = await Board.findById(id);
+
+  if (!board) {
+    res.status(404);
+    throw new Error("Board not found");
+  }
```

Comment:
This code is repetitive in the codebase, middleware helps keep our controllers leaner. I would create a middleware that handles checking if the board exists and if it doesn't we just respond with `Board not found` instead of having to do this over and over in controllers 👍🏻

Reply:
This sounds like a good idea. I will have to look into creating custom middleware for this as this is not something I've done before.

Reply:
For sure, feel free to reference my PR for teams 👍🏻